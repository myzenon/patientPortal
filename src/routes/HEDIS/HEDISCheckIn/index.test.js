import React from 'react'
import { HEDISCheckIn } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
        }
    }
    const wrapper = render(
        <HEDISCheckIn  { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks Next (Smart Question is completed) | go to Prepare Chapter 5 page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            smartQuestionsCompleted: true
        },
        push: (url) => {
            expect(url).toBe('/prepare-Chapter5')
            done()
        }
    }
    const wrapper = mount(
        <HEDISCheckIn  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').first().simulate('click')
    }, 0)
})

test('clicks Next (Smart Question is not completed) | go to Prepare Chapter 5 page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
        },
        push: (url) => {
            expect(url).toBe('/smart-questions/hedis/1')
            done()
        }
    }
    const wrapper = mount(
        <HEDISCheckIn  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').first().simulate('click')
    }, 0)
})
