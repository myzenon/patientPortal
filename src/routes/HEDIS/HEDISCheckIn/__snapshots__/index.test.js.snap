// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="row"
    >
      <div
        class="col-12 col-sm-4"
      >
        <img
          alt="hedis-checkin"
          src="hedis-checkin.png"
        />
      </div>
      <div
        class="col-12 col-sm-8"
      >
        <p>
          We're almost done!
        </p>
        <p>
          We only have a few questions left about your health conditions
        </p>
      </div>
    </div>
    <button
      class="button btn btn-secondary"
      type="button"
    >
      NEXT
    </button>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumberActive"
          >
            3
          </div>
        </div>
        <div
          class="stepActive"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumberActive"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
